### PR TITLE
feat(organizations): org-scoped group URLs, membership on join, and role-based visibility

### LINKED ISSUE #7646: Feature: Addition of nested group URLs in Organization
### Is your feature request related to a problem? Please describe

Groups that belong to an organization are currently only reachable at the top-level URL `/groups/:id`. The URL gives no indication that a group belongs to an organization, so there is no way to tell from the address (or from a shared link) which organization a group is part of, and no organization-scoped path for a group.

Now that groups can be created inside organizations, their URLs should be able to reflect that relationship.

### Describe the solution you'd like

Support organization-scoped group URLs alongside the existing top-level ones:

- `/organizations/:organization_id/groups/:id` - a group in the context of its organization
- `/groups/:id` - remains valid (existing links, bookmarks and shares must not break)

Both URLs stay active and resolve to the same group.

Handling a deleted organization:

Organizations use `dependent: :nullify` on groups, so deleting an organization leaves its groups intact but with `organization_id` set to nil. If someone then opens a nested URL for such a group (`/organizations/:organization_id/groups/:id`), the organization no longer exists, so the request must be resolved rather than 404:

- If the group no longer belongs to the organization in the URL (or the organization does not exist), redirect to the group's canonical top-level URL `/groups/:id`.
- If the group does belong to the organization, render normally.

This keeps old links working after an organization is deleted, instead of breaking them.

Also adds an organization-scoped groups index (`/organizations/:organization_id/groups`) listing the groups that belong to the organization.

### Describe alternatives you've considered

N/A

### Additional context

Project 6 Enterprise & Institutional Organization Features

### Are you working on this?

Yes

### PR DESCRIPTION
**Pull request by naman79820**
Fixes #7646

 

#### Describe the changes you have made in this PR -
This PR makes organization-owned groups consistent with organization membership, and scopes group visibility by role on the organization dashboard.

- Membership on join: When a user joins an organization-owned group via an invite link, they are also added to the organization as a basic member. This prevents users from belonging to an org's group without belonging to the org itself. Standalone groups are unaffected, and users who already have a role in the org (admin or mentor) keep it, they are not downgraded.
- Org-scoped group URLs: Invite and token-generation links for organization-owned groups are now organization-scoped (/organizations/:org/groups/:id/...). Standalone groups keep their existing flat URLs.
- Role-based group visibility: On the organization overview, org admins see all groups, while mentors and members see only the groups they are associated with (mentors see groups they own or mentor; members see groups they are a member of). An empty state is shown when a user has no groups.
- Authorization fix: Org admins can now view and manage organization-owned groups that other mentors created (previously blocked by a group-level access check that didn't account for org roles).

### Screenshots of the UI changes (If any) -

https://github.com/user-attachments/assets/90f292ba-e558-4359-99bc-94e22da35578

---

## Code Understanding and AI Usage

**Did you use AI generated code (ChatGPT, Claude, Copilot, etc.) in any part of this PR? **
- [x] No, I wrote all the code myself
- [ ] Yes, I used AI assistance (continue below)

**If you used AI assistance:**
- [ ] I have reviewed every single line of the AI-generated code
- [ ] I can explain the purpose and logic of each function/component I added
- [ ] I have tested edge cases and understand how the code handles them
- [ ] I have modified the AI output to follow this project's coding standards and conventions

**Explain your implementation approach:**
The problem: Organization-owned groups could have members who weren't part of the organization, which creates an inconsistent permission model. Also, the overview showed all groups to everyone regardless of role, and org admins couldn't manage groups other mentors created.

Membership on join: In the group_invite action, after adding the user to the group, I add them to the organization if the group belongs to one. I used find_or_create_by with the role set only in the creation block, so existing members keep their role (an admin joining a group stays an admin) and no duplicate memberships are created. I extracted this into a join_group_and_organization helper for readability. Standalone groups skip this entirely via an early return.

Group visibility: I scoped the overview's group query by role, admins get all org groups, and non-admins get groups they belong to or are the primary mentor of, scoped to that org. This keeps what a user sees consistent with what they can access.

Authorization fix: The check_show_access before-action used the base group policy, which only recognized a group's primary mentor so org admins were blocked from managing groups they didn't create. I made it organization-aware (like check_edit_access already was), routing org-groups through OrganizationGroupPolicy with a new show? method that allows org admins, assigned mentors, and group members.

Alternatives considered: For the invite URL, I considered keeping it flat since invite links are for outsiders, but scoped it to match the org context (the join action isn't membership-gated, so it doesn't block non-members). For mentor visibility, I chose "sees only their groups" over "sees all groups" for now, which can be broadened later.

---

## Checklist before requesting a review
- [x] I have added proper PR title and linked to the issue
- [x] I have performed a self-review of my code
- [x] **I can explain the purpose of every function, class, and logic block I added**
- [x] I understand why my changes work and have tested them thoroughly
- [x] I have considered potential edge cases and how my code handles them
- [x] If it is a core feature, I have added thorough tests
- [x] My code follows the project's style guidelines and conventions

---

Note: Please check **Allow edits from maintainers** if you would like us to assist in the PR.

## Summary by CodeRabbit

* **New Features**
  * Added organization dashboards with overview, members, and settings tabs.
  * Added branding, descriptions, social links, group listings, pagination, and management actions.
  * Added organization-scoped group pages, invitations, navigation, and redirects.
  * Added organization slug availability checks.
* **Bug Fixes**
  * Improved access control, protecting primary mentors and sole administrators.
  * Added validation and normalization for organization names and website links.
* **Documentation**
  * Added translations for organization dashboard content and navigation.

### CODE REVIEW

File: app/views/organizations/members.html.erb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<% content_for :title, t("organizations.members.page_title", organization: @organization.name) %>
+
+<%= render(OrganizationDashboardShellComponent.new(organization: @organization, active_tab: "members")) do |shell| %>
+  <% shell.with_tab_content do %>
+    <div class="org-dash-empty">
+      <i class="fa fa-users" aria-hidden="true"></i>
+      <p>Members management is coming soon.</p>
+    </div>
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/views/organizations/members.html.erb:8]: will update in member and settings pr

File: config/routes.rb [hunk 2]
```diff
@@ -25,10 +25,23 @@
...
+    end
     resources :organization_members, only: %i[create update destroy]
     delete :leave, to: "organization_members#leave"
-    resources :groups, only: %i[new create]
+    resources :groups, only: %i[new create show edit update] do
+      member do
+        get "invite/:token", to: "groups#group_invite", as: "invite"
+        put :generate_token
+      end
+    end
```
[the remark this answers is not in the document]
Comment (naman79820) [reply config/routes.rb:44]: The org's group listing already exists via overview, we dont need index

File: app/views/organizations/overview.html.erb [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+          <%= link_to new_organization_group_path(@organization), class: "btn primary-button d-inline-flex align-items-center gap-1" do %>
+            <i class="fa fa-plus" aria-hidden="true"></i> <%= t(".new_group_button") %>
+          <% end %>
+        <% end %>
+      </div>
+      <% if @groups.blank? %>
+        <div class="org-dash-empty text-center rounded-4 py-5 px-3 text-secondary">
+          <i class="fa fa-layer-group d-block mb-2 opacity-50" aria-hidden="true"></i>
+          <p class="m-0 fw-medium"><%= t("organizations.show.no_groups") %></p>
+        </div>
```
[the remark this answers is not in the document]
Comment (naman79820) [reply app/views/organizations/overview.html.erb:45]: recheck it is resolved @coderabbitai

**Comment by naman79820** [comment]
too much conflicts. Creating new pr.
